script.js

```diff
@@ -1,4 +1,275 @@
+var player1DiceRoll1 = 0;
+var player1DiceRoll2 = 0;
+var player2DiceRoll1 = 0;
+var player2DiceRoll2 = 0;
+var player1FinalDiceRollResult = 0;
+var player2FinalDiceRollResult = 0;
+var finalPlayer1AdvancedResults = 0;
+var finalPlayer2AdvancedResults = 0;
+var player1AdvanceRoll = [];
+var player2AdvanceRoll = [];
+var myOutputValue = "";
+var player1WinCount = 0;
+var player2WinCount = 0;
+var currentGameMode = "";
```

Comment:
Try to give the starting game mode a name as well. This is so that if you were explaining your code out loud, you can reference it by name.